lalrpop/src/main.rs

```diff
@@ -86,24 +85,17 @@ fn parse_args(mut args: Arguments) -> Result<Args, Box<dyn std::error::Error>> {
     })
 }
 
-fn main() {
-    main1().unwrap();
-}
-
-fn main1() -> Result<(), Box<dyn std::error::Error>> {
-    let mut stderr = std::io::stderr();
-    let mut stdout = std::io::stdout();
-
+fn main() -> Result<(), Box<dyn std::error::Error>> {
     let args = parse_args(Arguments::from_env())?;
 
     if args.flag_help {
-        writeln!(stdout, "{}", USAGE)?;
-        process::exit(0);
+        println!("{USAGE}");
```

Comment:
Beware that this is not equivalent to previous code. `println` panic in case of error, while `writeln` will bubble up the error. See https://github.com/BurntSushi/advent-of-code/issues/17#issuecomment-475443686. I'm not sure it makes a difference here, at least for now, as we are propagating the error to `main` anyway, but I wonder if we should keep the `writeln` by principle, so that if we one day we want to catch the error and do something else, we can?

Reply:
The linked comment applies here too, especially since printing isn't even the purpose of the application, and is only done for errors and the version/help strings.

Reply:
Did you mean "doesn't apply"?

Reply:
I'm referring to the last paragraph:

> println! is fine in short programs or where this sort of thing doesn't matter,
but I was trying to write idiomatic code, and IMO, idiomatic command line
programs don't panic.

Reply:
I still think the error message is different: one is panicking with a line number pointing to the inner code of `lalrpop`, while bubbling up an error would terminate with a `EXIT_FAILURE` and no mention of any rust source code? I haven't tested it. But I think the latter is nicer: a panic looks like something went horribly wrong and is a bug of `lalrpop`, while something like a broken pipe on printing is really not lalrpop's fault.

Reply:
This is a part of user interface. This change looks ok to me.